ui/src/app/applications/components/applications-list/applications-filter.tsx

```diff
@@ -33,9 +33,17 @@ export function getFilterResults(applications: Application[], pref: AppsListPref
             clusters:
                 pref.clustersFilter.length === 0 ||
                 pref.clustersFilter.some(
-                    selector =>
-                        (app.spec.destination.server && selector.includes(app.spec.destination.server)) ||
-                        (app.spec.destination.name && selector.includes(app.spec.destination.name))
+                    filterString =>
```

Comment:
I agree with idea @ciiay, thank you!

Just want to propose slightly different implementation:

```typescript
pref.clustersFilter.some(filterString => {
    const match = filterString.match('^.* [(](http.*)[)]$');
    if (match?.length === 2) {
        const [, url] = match;
        return url === app.spec.destination.server;
    } else {
        return app.spec.destination.name && minimatch(app.spec.destination.name, filterString);
    }
}),
```

It is pretty much the same but it extracts URL using regex and uses `minimatch` so that user could choose between exact and fuzzy much for cluster name